src/main/java/org/oscarehr/PMmodule/dao/ProgramDaoImpl.java

```
@@ -232,214 +296,270 @@ public List<Program> getPrograms() {
+        query.setParameter(1, genderType);
+        return query.getResultList();
     }
 
+    /**
+     * Saves a program.
+     * 
+     * @param program the Program object to save
+     * @throws IllegalArgumentException if the program is null
+     */
     @Override
     public void saveProgram(Program program) {
         if (program == null) {
             throw new IllegalArgumentException();
         }
         program.setLastUpdateDate(new Date());
 
-        // Adjusting flush mode
-        Session session = currentSession();
-        FlushMode previousFlushMode = session.getHibernateFlushMode();
-        session.setHibernateFlushMode(FlushMode.COMMIT);
-
         try {
-            getHibernateTemplate().saveOrUpdate(program);
+            entityManager.merge(program); // Use merge to save or update
+            if (log.isDebugEnabled()) {
+                log.debug("saveProgram: " + program.getId());
+            }
         } finally {
-            session.setHibernateFlushMode(previousFlushMode); // Restore the original flush mode
+            //swallow
         }
-        getHibernateTemplate().saveOrUpdate(program);
 
-        if (log.isDebugEnabled()) {
-            log.debug("saveProgram: " + program.getId());
-        }
     }
 
+    /**
+     * Removes a program by its ID.
+     * 
+     * @param programId the ID of the program to remove
+     * @throws IllegalArgumentException if the programId is null or less than or equal to 0
+     */
     @Override
     public void removeProgram(Integer programId) {
         if (programId == null || programId <= 0) {
             throw new IllegalArgumentException();
         }
-        try {
-            Object program = getHibernateTemplate().load(Program.class, programId);
 
-            getHibernateTemplate().delete(program);
+        Program program = entityManager.find(Program.class, programId);
+        if (program != null) {
+            entityManager.remove(program);
```

no try catch?

---

Same to above, we're not supposed to catch and swallow the exception. Instead, we should let it throw the exception so that the higher-level code can handle them.